include/psa/crypto_se_driver.h

```diff
@@ -685,6 +728,32 @@ typedef struct {
  */
 /**@{*/
 
+/* This type is documented in crypto.h. As far as drivers are concerned,
+ * this is an opaque type. */
+typedef struct psa_key_attributes_s psa_key_attributes_t;
+
+/** \brief A function that allocates a slot number for a key.
+ *
+ * This function is typically implemented as one or more calls to
+ * psa_drv_cb_find_free_slot(), with bounds determined by the key
+ * attributes and the secure element configuration.
+ *
+ * \param[in] attributes    Attributes of the key.
+ * \param[in] slot_usage    Slot usage data of the driver.
+ * \param[out] key_slot     Slot where the key will be stored.
+ *                          This must be a valid slot for a key of the
+ *                          chosen type. It must be unoccupied.
+ *
+ * \retval #PSA_SUCCESS
+ *         Success.
+ * \retval #PSA_ERROR_NOT_SUPPORTED
+ * \retval #PSA_ERROR_INSUFFICIENT_STORAGE
+ */
+typedef psa_status_t (*psa_drv_se_allocate_key_t)(
+    const psa_key_attributes_t *attributes,
+    const psa_drv_se_slot_usage_t *slot_usage,
+    psa_key_slot_number_t *key_slot);
```

Comment:
Should the policy also be provided here, so that the driver can avoid allocating a key that would be unsuitable for the desired use case? It'd stink to be an application and have to call allocate a bunch of times in a loop until you got allocated a key that worked with your desired usage policy.

Reply:
The policy _is_ provided. It's part of `attributes`.

Reply:
Excellent.